src/control/lib/control/auto.go

```diff
@@ -228,29 +234,33 @@ func (req *ConfigGenerateReq) genConfig(interfaces []*HostFabricInterface) *conf
 
 // checkNetwork scans fabric network interfaces and updates-in-place configuration
 // with appropriate fabric device details for all required numa nodes.
-func (req *ConfigGenerateReq) checkNetwork(ctx context.Context, resp *ConfigGenerateResp) error {
+//
+// Updates response config field in-place and returns number of cores per NUMA
+// node, which is the same on all hosts in the host set.
+func (req *ConfigGenerateReq) checkNetwork(ctx context.Context, resp *ConfigGenerateResp) (int, error) {
```

Comment:
It's kind of weird that a function named `checkNetwork()` is returning the number of cores per NUMA node, isn't it?

Out of curiosity, why not use the `netdetect.CoresPerNumaNode()` helper that you added?

Reply:
That would give you the cores on the client (the node that dmg is being run on), the helper needs to be run on the server which happens in NetworkScan request handler. Make sense?

Reply:
I understand why this is where the number of cores per NUMA node is coming from, but it still feels surprising to return it here. I think a minor refactoring would make things clearer. Here is one rough suggestion, you may be able to think of something better:

- add a new method req.buildConfig(resp *GenerateConfigResp) to encapsulate all of the logic:
  - req.getSingleNetworkSet
  - req.checkNetwork (now accepts req and fabricInfo, doesn't call getSingleNetworkSet itself anymore)
  - req.checkStorage
  - req.checkCPUs (fabricInfo.CoresPerNumaNode can now be passed in directly)

Reply:
this may make things clearer but the reason I didn't do this is that it makes testing awkward, mock invoker currently doesn't support returning different types of responses as far as I know so in the above example we will have to add another test for getSingleNetworkSet, but I can go down that route if preferred